mkosi/distributions/gentoo.py

```diff
@@ -35,40 +33,39 @@ def invoke_emerge(
             "--usepkg=y",
             "--keep-going=y",
             "--jobs",
-            "--load-average",
+            # Make sure the system stays somewhat usable by putting the max load average at 90%.
+            f"--load-average={0.9 * len(os.sched_getaffinity(0))}",
```

Comment:
how would this interact with [`portage`'s logic for `--jobs` and `--load-average`](https://github.com/gentoo/portage/blob/master/lib/portage/util/cpuinfo.py)?

Reply:
Not sure what you mean here? This makes sure gentoo doesn't use more than 90% of the allocated CPUs to keep the system somewhat responsive even during a build

Reply:
not sure, afk, but I think there is that logic baked into portage. this effects -j and -l in MAKEOPTS too not just emerge. I would leave it up to portage to handle this.
in my personal experience not passing anything here has only increased the load with 0 noticable effect on system's responsiveness.

Reply:
But the documentation for --load-average mentions that providing an empty value removes any previous assignment, so by setting an empty value we pretty much unset the load average.

Reply:
see the link I posted above to the code itself. and I wouldn't stop there, load_average is spread all over the code. documentation is most probably misleading. leaving it empty, from what I remember, tells portage to go parallel and if there is no value, find out number of cores etc via get_cpu_count() which actually calls getaffinity().

Reply:
Dropped these changes